educator/src/Dscp/Educator/Web/Student/Auth.hs

```diff
@@ -0,0 +1,74 @@
+{-# LANGUAGE DataKinds     #-}
+-- | Nessesary types and implementation for Student authenthication
+
+module Dscp.Educator.Web.Student.Auth
+       ( StudentAuth
+       , WithStudent (..)
+       , AuthData (..)
+       , GetStudents
+       ) where
+
+import Crypto.JOSE.JWK hiding (hash)
+import Crypto.JWT hiding (hash)
+import Data.Aeson
+import Servant.Auth.Server
+import Servant.Auth.Server.Internal.Class
+
+import Dscp.Crypto (PublicKey, AbstractPK (..), hash)
+import Dscp.Core (Address (..), Student)
+
+---------------------------------------------------------------------------
+-- Data types
+---------------------------------------------------------------------------
+
+-- | Custom authenthication type
+data StudentAuth
+
+-- | A type that has a Student attached to it
+data WithStudent a = WithStudent Student a
+
+-- | A type that the student has sent to authenthicate
+data AuthData = AuthData
```

Comment:
So we should figure out what is it exactly, right?

Reply:
Yes.

Reply:
Also I probably should change this comment because right now it doesn't make sense 😅